### PR TITLE
params: fix golint warnings

### PR DESCRIPTION
**Pull request by kielbarry**
Originally only intended to update params, but updating ChainId to ChainID required multiple changes in various other packages. ~5 golint warnings remain in params.

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -101,7 +102,7 @@ var (
 // that any network, identified by its genesis block, can have its own
 // set of configuration options.
 type ChainConfig struct {
-	ChainId *big.Int `json:"chainId"` // Chain id identifies the current chain and is used for replay protection
+	ChainID *big.Int `json:"chainID"` // Chain id identifies the current chain and is used for replay protection
```
Comment (karalabe) [inline params/config.go]: This one changes our genesis format, please revert. Yes, it's wrong, but it's part of the API, so if you uppercase it, Go will refuse to parse the jsons.

File: params/config.go [hunk 2]
```diff
@@ -172,27 +173,32 @@ func (c *ChainConfig) IsHomestead(num *big.Int) bool {
 	return isForked(c.HomesteadBlock, num)
 }
 
-// IsDAO returns whether num is either equal to the DAO fork block or greater.
+// IsDAOFork returns whether num is either equal to the DAO fork block or greater.
 func (c *ChainConfig) IsDAOFork(num *big.Int) bool {
 	return isForked(c.DAOForkBlock, num)
 }
 
+// IsEIP150 returns whether num is either equal to the IsEIP150 fork block or greater.
```
Comment (karalabe) [inline params/config.go]: I don't think you need `Is` in the second instance of `IsEIP150`. (same for all other fork docs below).

File: params/version.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 	"fmt"
 )
 
+// Version components of the current release.
```
Comment (karalabe) [inline params/version.go]: This is really not that useful.

File: params/config.go (hunk 1 above)
Comment (karalabe) [reply params/config.go]: Just to clarify, the field name is fine as you changed it, just the json tag needs to be reverted.

Comment (kielbarry) [reply params/config.go]: Great ok - updated.

File: params/config.go (hunk 2 above)
Comment (kielbarry) [reply params/config.go]: You're right - I've updated to be more sensical.

File: params/version.go (hunk 3 above)
Comment (kielbarry) [reply params/version.go]: Removed

**Pull request merged**